File: app/src/main/java/com/makingiants/todayhistory/utils/DateManager.java
package com.makingiants.todayhistory.utils;

import android.os.Parcel;
import android.os.Parcelable;
import java.util.Calendar;

public class DateManager implements Parcelable {
  Calendar mCalendar;

  public DateManager() {
    mCalendar = Calendar.getInstance();
  }

  public int getTodayDay() {
    return mCalendar.get(Calendar.DAY_OF_MONTH);
  }

  public int getTodayMonth() {
    Calendar cal = Calendar.getInstance();
    return mCalendar.get(Calendar.MONTH);
  }

  //<editor-fold desc="Parcelable">
  @Override
  public int describeContents() {
    return 0;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeSerializable(this.mCalendar);
  }

  protected DateManager(Parcel in) {
    this.mCalendar = (Calendar) in.readSerializable();
  }

  public static final Parcelable.Creator<DateManager> CREATOR =
      new Parcelable.Creator<DateManager>() {
        public DateManager createFromParcel(Parcel source) {
          return new DateManager(source);
        }

        public DateManager[] newArray(int size) {
          return new DateManager[size];
        }
      };
  //</editor-fold>
}

File: app/src/main/kotlin/com/makingiants/todayhistory/utils/DateManager.kt
package com.makingiants.todayhistory.utils

import android.os.Parcel
import android.os.Parcelable
import java.util.*

open class DateManager : Parcelable {
    internal var mCalendar: Calendar

    constructor() {
        mCalendar = Calendar.getInstance()
    }

    open fun getTodayDay(): Int = mCalendar.get(Calendar.DAY_OF_MONTH)
    open fun getTodayMonth(): Int = mCalendar.get(Calendar.MONTH)

    //<editor-fold desc="Parcelable">
    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeSerializable(this.mCalendar)
    }

    protected constructor(`in`: Parcel) {
        this.mCalendar = `in`.readSerializable() as Calendar
    }

    companion object {
        val CREATOR: Parcelable.Creator<DateManager> = object : Parcelable.Creator<DateManager> {
            override fun createFromParcel(source: Parcel): DateManager {
                return com.makingiants.todayhistory.utils.DateManager(source)
            }

            override fun newArray(size: Int): Array<out DateManager?> = arrayOfNulls(size)

        }
    }
    //</editor-fold>
}
